### DISCUSSION TITLE
Google results for date-fns functions return out-dated materials

### DISCUSSION THREAD

**Discussion by ChristopherWirtOfficial** [js-fns/date-fns#2582] [General]
Every time I go to search for the API specs of a date-fns function (looking at you, format) I end up at the version 1.9.0 documentation. According to npm numbers, this version is neither the most recent nor the most popular (1 download this past week). I understand that 1.30 is still a very popular version of the package, but the average user will either be aware that they're on a legacy version (1.X) and ensure they're on that version of the API spec, or they'll probably not know that there are popular older versions, in which case they'll have no idea the docs aren't showing the latest version by default. 

All of that being said, I obviously don't know if this is intentional or not. It could just be that the docs website needs to be updated so that search engines will prefer the latest versions of the functions for search results. It's pretty frustrating for a google result to bring you to a page that's been out of date for nearly 4 years.

**Comment by tan75** [comment]
@Chrisazy thanks for reporting this. Much appreciated!
Could you please provide the exact wordings you were using in your google search?

**Reply by fturmel** [reply]
@tan75 Searching for "date-fns format", the second link I get does go to the 1.9.0 version of the docs (see the hover URL hint at the bottom).

<img width="1265" alt="Screen Shot 2021-08-10 at 7 41 13 AM" src="https://user-images.githubusercontent.com/4009209/128861052-d5d4c28f-bb2d-4aed-9d1b-6133064fe863.png">

The same link surfaces if you search for "date-fns docs", and that other nested one goes to v2.16.1 of `startOfToday`.

<img width="1002" alt="Screen Shot 2021-08-10 at 7 45 40 AM" src="https://user-images.githubusercontent.com/4009209/128861444-2d0aa6e3-0465-4b22-8d9e-f78e3bd44ba4.png">

**Reply by ChristopherWirtOfficial** [reply]
"date-fns format" is what I typed to get to [this URL](https://date-fns.org/v1.9.0/docs/format). I usually type similar queries and have problems a lot like fturmel pointed out in their replies.

**Comment by fturmel** [comment]
There certainly are a number of issues with the documentation from an SEO perspective, it should probably be re-architected.

It doesn't seem like Googlebot can properly crawl the docs contents to begin with, so there are no deeplinks when you search for the documentation of a function. ex: Googling "date-fns docs parseISO", I get these awful results:
1. the main docs entry page (https://date-fns.org/docs/, which redirects to https://date-fns.org/docs/Getting-Started)
2. v1.29.0 docs link to the parse function (https://date-fns.org/v1.29.0/docs/parse)
3. the upgrade guide markdown file from github (https://github.com/date-fns/date-fns/blob/master/docs/upgradeGuide.md)
4. a bunch of stackoverflow links

The URLs containing the full semver certainly brings some issues too. In my opinion, it would be better to have 1.x and 2.x "evergreen" urls, and also use those as canonical link for existing semver urls. These are the ones you'd want search engine crawlers and most documentation readers to use / share.

A more prominent warning on pages that are not the latest version would also help, especially for users coming from old Stackoverflow answers or github issues/discussions that contained semver links.

Projects like Jest and Laravel seem to do a good job at these specifically, perhaps their approach could be referenced.

**Comment by tan75** [comment]
Thanks for the feedback and ideas everyone! Appreciated! 
There is a lot of work has to be done over the docs website and this will take some time.
We will certainly schedule some SEO improvements and hopefully this issue will be resolved.
